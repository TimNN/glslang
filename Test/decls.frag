#version 120

int a;
int b, c;
int d1 = 1;
int e2 = 2, f;
int g, h3 = 3;

int i4[4];
int j, k5[5];
int m6[6], m7[7];
int n8[8], p;

int ii4[4] = int[](1, 2, 3, 4);
int ij, ik5[5] = int[](5, 6, 7, 8, 9);
int im2[2] = int[](10, 11), im3[3] = int[](12, 13, 14);
int in8[4] = int[](21, 22, 23, 24), ip;

void vi4[4] = int[](1, 2, 3, 4);
void vj, vk5[5] = int[](5, 6, 7, 8, 9);
void vm2[2] = int[](10, 11), vm3[3] = int[](12, 13, 14);
void vn8[4] = int[](21, 22, 23, 24), vp;

int gl_vi4[4] = int[](1, 2, 3, 4);
int gl_vj, gl_vk5[5] = int[](5, 6, 7, 8, 9);
int gl_vm2[2] = int[](10, 11), gl_vm3[3] = int[](12, 13, 14);
int gl_vn8[4] = int[](21, 22, 23, 24), gl_vp;

void main()
{
    while (bool cond = b < c);
    while (int icond = b);
    while (bool gl_cond = b < c);
}
