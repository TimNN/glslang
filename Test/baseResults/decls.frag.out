ERROR: 0:19: 'vi4' : illegal use of type 'void' 
ERROR: 0:20: 'vj' : illegal use of type 'void' 
ERROR: 0:20: 'vk5' : illegal use of type 'void' 
ERROR: 0:21: 'vm2' : illegal use of type 'void' 
ERROR: 0:21: 'vm3' : illegal use of type 'void' 
ERROR: 0:22: 'vn8' : illegal use of type 'void' 
ERROR: 0:22: 'vp' : illegal use of type 'void' 
ERROR: 0:24: 'gl_' : reserved built-in name 
ERROR: 0:25: 'gl_' : reserved built-in name 
ERROR: 0:25: 'gl_' : reserved built-in name 
ERROR: 0:26: 'gl_' : reserved built-in name 
ERROR: 0:26: 'gl_' : reserved built-in name 
ERROR: 0:27: 'gl_' : reserved built-in name 
ERROR: 0:27: 'gl_' : reserved built-in name 
ERROR: 0:32: '' : boolean expression expected 
ERROR: 0:33: 'gl_' : reserved built-in name 
ERROR: 16 compilation errors.  No code generated.

ERROR: node is still EOpNull!
0:5  Sequence
0:5    move second child to first child (int)
0:5      'd1' (int)
0:5      1 (const int)
0:6  Sequence
0:6    move second child to first child (int)
0:6      'e2' (int)
0:6      2 (const int)
0:7  Sequence
0:7    move second child to first child (int)
0:7      'h3' (int)
0:7      3 (const int)
0:14  Sequence
0:14    move second child to first child (4-element array of int)
0:14      'ii4' (4-element array of int)
0:14      1 (const int)
0:14      2 (const int)
0:14      3 (const int)
0:14      4 (const int)
0:15  Sequence
0:15    move second child to first child (5-element array of int)
0:15      'ik5' (5-element array of int)
0:15      5 (const int)
0:15      6 (const int)
0:15      7 (const int)
0:15      8 (const int)
0:15      9 (const int)
0:16  Sequence
0:16    move second child to first child (2-element array of int)
0:16      'im2' (2-element array of int)
0:16      10 (const int)
0:16      11 (const int)
0:16    move second child to first child (3-element array of int)
0:16      'im3' (3-element array of int)
0:16      12 (const int)
0:16      13 (const int)
0:16      14 (const int)
0:17  Sequence
0:17    move second child to first child (4-element array of int)
0:17      'in8' (4-element array of int)
0:17      21 (const int)
0:17      22 (const int)
0:17      23 (const int)
0:17      24 (const int)
0:24  Sequence
0:24    move second child to first child (4-element array of int)
0:24      'gl_vi4' (4-element array of int)
0:24      1 (const int)
0:24      2 (const int)
0:24      3 (const int)
0:24      4 (const int)
0:25  Sequence
0:25    move second child to first child (5-element array of int)
0:25      'gl_vk5' (5-element array of int)
0:25      5 (const int)
0:25      6 (const int)
0:25      7 (const int)
0:25      8 (const int)
0:25      9 (const int)
0:26  Sequence
0:26    move second child to first child (2-element array of int)
0:26      'gl_vm2' (2-element array of int)
0:26      10 (const int)
0:26      11 (const int)
0:26    move second child to first child (3-element array of int)
0:26      'gl_vm3' (3-element array of int)
0:26      12 (const int)
0:26      13 (const int)
0:26      14 (const int)
0:27  Sequence
0:27    move second child to first child (4-element array of int)
0:27      'gl_vn8' (4-element array of int)
0:27      21 (const int)
0:27      22 (const int)
0:27      23 (const int)
0:27      24 (const int)
0:29  Function Definition: main( (void)
0:29    Function Parameters: 
0:31    Sequence
0:31      Loop with condition tested first
0:31        Loop Condition
0:31        move second child to first child (bool)
0:31          'cond' (bool)
0:31          Compare Less Than (bool)
0:31            'b' (int)
0:31            'c' (int)
0:31        No loop body
0:32      Loop with condition tested first
0:32        Loop Condition
0:32        move second child to first child (int)
0:32          'icond' (int)
0:32          'b' (int)
0:32        No loop body
0:33      Loop with condition tested first
0:33        Loop Condition
0:33        move second child to first child (bool)
0:33          'gl_cond' (bool)
0:33          Compare Less Than (bool)
0:33            'b' (int)
0:33            'c' (int)
0:33        No loop body
0:?   Linker Objects
0:?     'a' (int)
0:?     'b' (int)
0:?     'c' (int)
0:?     'd1' (int)
0:?     'e2' (int)
0:?     'f' (int)
0:?     'g' (int)
0:?     'h3' (int)
0:?     'i4' (4-element array of int)
0:?     'j' (int)
0:?     'k5' (5-element array of int)
0:?     'm6' (6-element array of int)
0:?     'm7' (7-element array of int)
0:?     'n8' (8-element array of int)
0:?     'p' (int)
0:?     'ii4' (4-element array of int)
0:?     'ij' (int)
0:?     'ik5' (5-element array of int)
0:?     'im2' (2-element array of int)
0:?     'im3' (3-element array of int)
0:?     'in8' (4-element array of int)
0:?     'ip' (int)
0:?     'gl_vi4' (4-element array of int)
0:?     'gl_vj' (int)
0:?     'gl_vk5' (5-element array of int)
0:?     'gl_vm2' (2-element array of int)
0:?     'gl_vm3' (3-element array of int)
0:?     'gl_vn8' (4-element array of int)
0:?     'gl_vp' (int)

